src/components/LinkRenderer/LinkRenderer.tsx

```diff
@@ -1,19 +1,42 @@
 import { memo } from 'react';
+import s from './LinkRenderer.module.css';
 
-export const LinkRenderer = memo(function (props: any) {
+// types from react-markdown expected types
+type ReactMemoElementWrapProps = React.DetailedHTMLProps<
+  React.AnchorHTMLAttributes<HTMLAnchorElement>,
+  HTMLAnchorElement
+>;
+
+export const LinkRenderer = memo(function (props: ReactMemoElementWrapProps) {
   return (
-    <a
-      href={props.href}
-      target="_blank"
-      rel="noreferrer"
-      onClick={stopPropagation}
-    >
+    <a href={props.href} target="_blank" rel="noreferrer" onClick={stopPropagation}>
       {props.children}
     </a>
   );
 });
+
+export const LinkRendererShort = memo(function ({
+  children: linkArr,
+}: ReactMemoElementWrapProps) {
+  return (
+    <div className={s.linkWidthWrap}>
+      <div className={s.linkOverflowWrap}>
+        <a
+          className={s.link}
+          target="_blank"
+          rel="noreferrer"
+          data-truncate={linkArr?.[0].slice(-12)}
```

Comment:
why -12? I would like to see any comments there

Reply:
okay!
it was suggested by the task i'm implementing